<?xml version="1.0" encoding="utf-8"?>
<launch>

  <arg name="model" default="Hydra_hex" /> 
  <arg name="start_angle" default="1.0472" /> <!-- penta:1.256, hex: 1.0471-->
  <arg name="goal_angle" default="-1.0472" /> 

  <!-- Debug Info -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <node name="motion_planning_node" pkg="hydra_gap_passing" type="motion_planning_node" respawn="false" launch-prefix="$(arg launch_prefix)" output="screen">
    <rosparam command="load" file="$(find hydra_moveit_config)/config/kinematics.yaml"/>
    <param name="planning_plugin" value="ompl_interface/OMPLPlanner"/>
    <rosparam command="load" file="$(find hydra_moveit_config)/config/ompl_planning.yaml"/>
    <param name="coefficient_rate" value="0.05"/> 
    <param name="planning_mode" value="0"/> 
    <param name="motion_sequence_rate" value="50"/>
    <param name="ompl_mode" value="0"/> <!-- RRT START:0 -->

    <param name="gap_x_offset" value="0.45"/> 
    <param name="gap_y_offset" value="0"/> 

    <param name="gap_left_width" value="0.03"/> 
    <param name="gap_right_width" value="0.029"/>  

    <param name="start_state_joint1" value="$(arg start_angle)"/> 
    <param name="start_state_joint2" value="$(arg start_angle)"/> 
    <param name="start_state_joint3" value="$(arg start_angle)"/> 
    <param name="start_state_joint4" value="$(arg start_angle)"/> 
    <param name="start_state_joint5" value="$(arg start_angle)"/> 

    <param name="goal_state_joint1" value="$(arg goal_angle)"/> 
    <param name="goal_state_joint2" value="$(arg goal_angle)"/> 
    <param name="goal_state_joint3" value="$(arg goal_angle)"/> 
    <param name="goal_state_joint4" value="$(arg goal_angle)"/> 
    <param name="goal_state_joint5" value="$(arg goal_angle)"/> 


    <param name="state_validity_check_res" value="0.03"/>
    <param name="valid_segment_count_factor" value="20"/> <!--20-->

    <!-- for transform controller -->
    <rosparam file="$(find hydra_transform_control)/config/hydrax/$(arg model).yaml" command="load" />
    <param name="debug_log" type="bool" value="false"/>

    <param name="solving_time_limit" type="double" value="3600"/>

    <param name="semi_stable_cost" type="double" value="1.0"/>
    <param name="full_stable_cost" type="double" value="0.1"/>

    <param name="length_opt_weight" type="double" value="1.0"/>
    <param name="stability_opt_weight" type="double" value="0"/>

    <param name="stability_cost_thre" type="double" value="1.2" />

    <param name="log_flag" type="bool" value="true" />

    <!-- for motion control -->
    <param name="play_log_path" type="bool" value="true"/> 
    <param name="file_name" value="$(find hydra_gap_passing)/path_log/planning_log_hydra_pendagon_thre_0p12.txt" />

    <remap from="/hydra/joints_ctrl" to="/joint_states" /> 

  </node>
</launch>
